ocs_ci/ocs/jenkins.py

```diff
@@ -0,0 +1,322 @@
+        return build_list
+
+    def create_project_names(self):
+        """
+        Create project names
+
+        """
+        self.projects = []
+        for project_id in range(1, self.num_of_projects + 1):
+            self.projects.append('myjenkins-' + str(project_id))
+
+    def create_ocs_jenkins_template(self):
+        """
+
+        Create OCS Jenkins Template
+        """
+        log.info("Create Jenkins Template, jenkins-persistent-ocs")
+        ocp_obj = OCP(namespace='openshift', kind='template')
+        tmp_dict = ocp_obj.get(
+            resource_name='jenkins-persistent', out_yaml_format=True
+        )
+        tmp_dict['labels']['app'] = 'jenkins-persistent-ocs'
+        tmp_dict['labels']['template'] = 'jenkins-persistent-ocs-template'
+        tmp_dict['metadata']['name'] = 'jenkins-persistent-ocs'
+        tmp_dict['objects'][1]['metadata']['annotations'] = {
+            'volume.beta.kubernetes.io/storage-class': 'ocs-storagecluster-ceph-rbd'
+        }
+        tmp_dict['objects'][2]['spec']['template']['spec']['containers'][0]['env'].append(
+            {'name': 'JAVA_OPTS', 'value': '${JAVA_OPTS}'})
+        tmp_dict['parameters'][4]['value'] = '10Gi'
+        tmp_dict['parameters'].append({
+            'description': "Override jenkins options to speed up slave spawning",
+            'displayName': 'Override jenkins options to speed up slave spawning',
+            'name': 'JAVA_OPTS',
+            'value': "-Dhudson.slaves.NodeProvisioner.initialDelay=0 "
+                     "-Dhudson.slaves.NodeProvisioner.MARGIN=50 -Dhudson."
+                     "slaves.NodeProvisioner.MARGIN0=0.85"
+        })
+        ocs_jenkins_template_obj = OCS(**tmp_dict)
+        ocs_jenkins_template_obj.create()
+
+    def get_builds_logs(self):
+        """
+
+        Get builds logs and print them on table
+        """
+        log.info('Print builds results')
+        build_table = PrettyTable()
+        build_table.field_names = ["Project", "Build", "Duration"]
+        for build in self.build_completed:
+            ocp_obj = OCP(namespace=build[1], kind='build')
+            output_build = ocp_obj.describe(resource_name=build[0])
+            list_build = output_build.split()
+            index = list_build.index('Duration:')
+            build_table.add_row([build[1], build[0], list_build[index + 1]])
+        log.info(f'\n{build_table}\n')
+
+    def cleanup(self):
+        """
+        Clean up
+
+        """
+        for project in self.projects:
+            log.info(f"Delete Jenkins project: {project}")
+            ocp_obj = OCP(namespace=project)
+            ocp_obj.delete_project(project)
+
+        log.info("Delete Jenkins Template")
+        ocp_obj = OCP()
+        cmd = "delete template.template.openshift.io/jenkins-persistent-ocs -n openshift"
+        ocp_obj.exec_oc_cmd(command=cmd, out_yaml_format=False)
+        # Wait for the resources to delete
+        time.sleep(120)
```

Comment:
please use ocp.wait_for_delete()

Reply:
The "delete" command works, but cluster doesn't actually delete(in same time) the resource and there is leftover issue

Reply:
will be tracked with https://github.com/red-hat-storage/ocs-ci/issues/2417